src/rest.rs

```diff
@@ -36,29 +37,71 @@ impl Client {
         Ok(Client {
             config,
             address,
-            token: token,
+            token,
             rest: client,
         })
     }
 
-    pub async fn req<T: Serialize + ?Sized>(
+    pub async fn req<T: Serialize + ?Sized, Q: Serialize + ?Sized>(
         &self,
         url: String,
         method: Method,
         body: Option<&T>,
+        query: Option<&Q>,
+        headers: Option<HeaderMap>,
     ) -> Result<Response, BrigadeError> {
-        let token = self.token.clone().unwrap();
-        let req = self.rest.request(method, &url).bearer_auth(token);
-        let res = match body {
-            Some(body) => req.json(body).send().await?,
-            None => req.send().await?,
-        };
+        let req = self.create_req(url, method, body, query, headers);
 
+        let res = req.send().await?;
         Ok(res)
     }
+
+    pub async fn req_with_basic_auth<T: Serialize + ?Sized, Q: Serialize + ?Sized>(
```

Comment:
Generally, you have a `Client` builder (different from the request builder) that allows you to configure auth, https, etc. Consider having `basic_auth` and `root_auth` methods on that builder that returns a `Client`, particularly with how big this is getting. It also allows a user to configure any default headers they need to send (if you expose that method)